src/catalog/src/kvbackend/manager.rs

```diff
@@ -217,29 +237,52 @@ impl CatalogManager for KvBackendCatalogManager {
             return Ok(Some(table));
         }
 
-        let key = TableNameKey::new(catalog, schema, table_name);
-        let Some(table_name_value) = self
-            .table_metadata_manager
-            .table_name_manager()
-            .get(key)
-            .await
-            .context(TableMetadataManagerSnafu)?
-        else {
-            return Ok(None);
+        let init = async {
+            let table_name_key = TableNameKey::new(catalog, schema, table_name);
+            let Some(table_name_value) = self
+                .table_metadata_manager
+                .table_name_manager()
+                .get(table_name_key)
+                .await
+                .context(TableMetadataManagerSnafu)?
+            else {
+                return TableCacheNotGetSnafu {
+                    key: table_name_key.to_string(),
+                }
+                .fail();
+            };
+            let table_id = table_name_value.table_id();
+
+            let Some(table_info_value) = self
+                .table_metadata_manager
+                .table_info_manager()
+                .get(table_id)
+                .await
+                .context(TableMetadataManagerSnafu)?
+                .map(|v| v.into_inner())
+            else {
+                return TableCacheNotGetSnafu {
+                    key: table_name_key.to_string(),
+                }
+                .fail();
+            };
+            make_table(table_info_value)
         };
-        let table_id = table_name_value.table_id();
 
-        let Some(table_info_value) = self
-            .table_metadata_manager
-            .table_info_manager()
-            .get(table_id)
+        match self
+            .table_cache
+            .try_get_with_by_ref(&format_full_table_name(catalog, schema, table_name), init)
```

Comment:
Looks like `AsyncCache` lacks an `AsyncCache::optionally_try_get_with_by_ref` that ignore the `None` value returned by `init` closure, just like [`AsyncCache::optionally_get_with_by_ref`](https://docs.rs/moka/0.12.1/moka/future/struct.Cache.html#method.optionally_get_with_by_ref) so that we don't have to match the error variants 😓